pkg/mimirtool/commands/backfill.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/grafana/mimir/pkg/mimirtool/client"
+)
+
+type BackfillCommand struct {
+	clientConfig client.Config
+	blocks       blockList
+}
+
+type blockList []string
+
+func (l *blockList) Set(value string) error {
+	st, err := os.Stat(value)
+	if err != nil {
+		return fmt.Errorf("directory %q doesn't exist", value)
+	}
+	if !st.IsDir() {
+		return fmt.Errorf("%q must be a directory", value)
+	}
+	*l = append(*l, value)
+	return nil
+}
+
+func (l blockList) String() string {
+	return strings.Join(l, ",")
+}
+
+func (l blockList) IsCumulative() bool {
+	return true
+}
+
+func (c *BackfillCommand) Register(app *kingpin.Application, envVars EnvVarNames) {
+	cmd := app.Command("backfill", "Upload metrics blocks to Grafana Mimir.")
+	cmd.Action(c.backfill)
+	cmd.Arg("block", "block to upload").Required().SetValue(&c.blocks)
+	cmd.Flag("address", "Address of the Grafana Mimir cluster").Required().StringVar(&c.clientConfig.Address)
+	cmd.Flag("id", "Grafana Mimir tenant ID").Required().StringVar(&c.clientConfig.ID)
```

Comment:
We removed that, as we thought it's confusing to have both `id` and `user`. Why are both needed, and what is the difference between them?

Reply:
The ID is used to set `X-Scope-OrgID`, or as a fallback for auth user. The user is just used for auth. In this PR I would get consistent with other command, then we can open a PR to further discuss it because I agree it somewhat looks redundant.

Also, just looking at the command help, id help is "Grafana Mimir tenant ID" and nothing makes me thing it will also be used as basic auth username.